clients/pkg/promtail/promtail.go

```diff
@@ -61,37 +72,62 @@ func New(cfg config.Config, metrics *client.Metrics, dryRun bool, opts ...Option
 		}
 		o(promtail)
 	}
+	err := promtail.reloadConfig(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	server, err := server.New(cfg.ServerConfig, promtail.logger, promtail.targetManagers, cfg.String())
+	if err != nil {
+		return nil, err
+	}
+	promtail.server = server
+	promtail.newConfig = newConfig
 
-	cfg.Setup(promtail.logger)
+	return promtail, nil
+}
+
+func (p *Promtail) reloadConfig(cfg *config.Config) error {
```

Comment:
I think we should add a metric to show how many times the config was reloaded, and whether it was successful or not

Reply:
done